libminifi/include/utils/MinifiConcurrentQueue.h

```diff
@@ -87,6 +91,10 @@ class ConcurrentQueue {
     queue_.emplace_back(std::forward<Args>(args)...);
   }
 
+  LockedConcurrentQueue<T> lock() {
```

Comment:
Why would you do this? If you really want to extend the functionality of the class, instead of adding friend accessors, make your queue inherit from the `ConcurrentQueue`.

Reply:
Because `LockedConcurrentQueue` **is not a** `ConcurrentQueue`. It's just a utility class that holds a lock to a `ConcurrentQueue`'s mutex and provides access to the underlying queue until it goes out of scope. So instead of locking the mutex with every operation you can do multiple operations with a single locking. I'm open to a better name for the class.